moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -612,33 +614,74 @@ class KinematicsBase
   template <typename T>
   inline bool lookupParam(const std::string& param, T& val, const T& default_val) const
   {
-    ros::NodeHandle pnh("~");
-    if (pnh.hasParam(group_name_ + "/" + param))
-    {
-      val = pnh.param(group_name_ + "/" + param, default_val);
-      return true;
+    // ros::NodeHandle pnh("~");
+
+    auto node = rclcpp::Node::make_shared("lookupparam");
+
+    auto parameters_lookup = std::make_shared<rclcpp::SyncParametersClient>(node);
+
+    auto groupname_param = parameters_lookup->get_parameters({group_name_ + "/" + param});
+
+    for (auto & parameter : groupname_param) {
+        if (!parameter.get_name().compare(group_name_ + "/" + param)) {
```

Comment:
Please clang format these PR's so that we don't have a huge clang PR later. We use the following bash function to automatically run `clang-format-3.9` on the entire moveit repo.

```bash!
# Setup:
#   sudo apt install clang-format-3.9

function console_red {
    echo -n -e "\e[00;31m"
}

function console_nored {
    echo -n -e "\e[00m"
}

alias clang-format="clang-format-3.9"

function clang_this_directory_custom # recursive
{
    # Make sure no changes have occured in repo
    if ! git diff-index --quiet HEAD --; then
        # changes
        console_red
        read -p "You have uncommitted changes, are you sure you want to continue? (y/n)" resp
        console_nored

        if [ "$resp" = "y" ]; then
            echo "Formatting..."
        else
            return -1
        fi
    fi

    find . -name '*.h' -or -name '*.hpp' -or -name '*.cpp' -o -name '*.c' -o -name '*.cc' | xargs clang-format-3.9 -i -style=file $1
}
```